### PR TITLE
Error with Multiple Projects of the Same Project on One Log

### PR DESCRIPTION
**Pull request by dhuey**
On one log if I Add Project and select the same project for the first project and second project, the system errors.

### CODE REVIEW

File: app/assets/javascripts/logs.coffee [hunk 1]
```diff
@@ -70,6 +72,19 @@ class Log
 
     updateTotalHours()
 
+    projectChangeListener = ->
+      $("select[id$='_project_id']").on "change", distinctProjects
+      $("select[id$='_project_id']").each distinctProjects # Doesn't work yet.
+
+    distinctProjects = ->
+      console.log("hit it")
+      $("select[id$='_project_id']").children().removeAttr 'disabled'
```
Comment (MicahBrown) [inline app/assets/javascripts/logs.coffee]: `$("select[id$='_project_id']")` is called multiple times and can be refactored into a variable

File: app/assets/javascripts/logs.coffee [hunk 2]
```diff
@@ -70,6 +72,19 @@ class Log
...
+      $("select[id$='_project_id']").each ->
+        if $(this).val() != ''
```
Comment (MicahBrown) [inline app/assets/javascripts/logs.coffee]: `unless $(this).val() == ''` also works in coffeescript

File: app/assets/javascripts/logs.coffee [hunk 3]
```diff
@@ -70,6 +72,19 @@ class Log
...
+          $("select[id$='_project_id']").not($(this)).find('option[value=' + $(this).val() + ']').attr 'disabled', 'disabled'
+
+    setTimeout projectChangeListener, 2000
```
Comment (MicahBrown) [inline app/assets/javascripts/logs.coffee]: What's the `setTimeout` for?

Comment (dhuey) [reply app/assets/javascripts/logs.coffee]: We used it in SQSS. I assumed it was to start the listener after the page loads.

**Comment by dhuey** [comment]
> Do you clean up the disabled options when they deselect a project that was selected? I don't see that handled anywhere

This gets handled in the `distinctUsers` function. On line 81 (of the previous commit), it removes any 'disabled' attributes, then reassigns it based upon what's currently selected

**Pull request merged**